server/index.js

```diff
@@ -0,0 +1,39 @@
+const express = require('express');
+const cors = require('cors');
+const cookieParser = require('cookie-parser');
+const dotenv = require('dotenv');
+const axios = require('axios').default;
+
+dotenv.config({ path: './.env' });
+
+const app = express();
+const port = process.env.PORT;
+
+app.use(cors({ credentials: true, origin: 'http://localhost:9020' }));
+
+app.use(express.json());
+app.use(cookieParser());
+
+app.post('/auth', async (req, res) => {
+  try {
+    const response = await axios.post(
+      'https://pfa.foreca.com/api/v1/authorize/token',
+      req.body
+    );
+    const { access_token: token, expires_in: expiresIn } = response.data;
+
+    res
+      .cookie('token', token, {
+        expires: new Date(Date.now() + expiresIn * 1000),
+        httpOnly: true,
+        secure: req.secure || req.headers['x-forwarded-proto'] === 'https',
+      })
+      .json({ status: 'success', token });
+  } catch (err) {
+    res.json({ status: 'fail', error: err });
+  }
```

Comment:
can you please explain flow of getting access_token (how often it is requested, where exactly it is stored, etc)?

Reply:
It is called once, when the app loads in useEffect hook with empty [] dependencies array in the App component. And it is stored in redux store calling dispatch(setCredentials(res.data.token)); Although, now I think I probably could call hook on [auth.token] dependency?

Reply:
ok, 2 more questions:
1. why do you use cookie-parser library?
2. if your app will be opened in two separate tabs, will it generate tokens twice?

Reply:
2. I guess it will. Since I'm not storing token in localStorage, as it is not a good idea, two separate tabs would work as two separate apps and on refresh would generate token, and it would not matter if I check whether token exists in redux store. And I can't read cookie, as it is set "httpOnly: true".
1. I am not actually, but maybe I should use it. Maybe I should not send token as a response in handleLogin fc on frontend, but call another fc, say checkAuthState, where I would do another request to server, and in the server I would use cookieParser to get req.cookies and then send the token?